pkg/kube/inject/webhook.go

```diff
@@ -1163,13 +1174,25 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		// application container's value. At the same time, if user explicitly configures a RunAsUser in the istio-proxy
 		// container which is different to the application container's value, that setting is still honored.
 		if sideCarProxy := FindSidecar(params.pod); sideCarProxy != nil && sideCarProxy.SecurityContext != nil {
-			if isSidecarUserMatchingAppUser(params.pod.Spec.Containers) {
+			if isSidecarUserMatchingAppUser(params.pod) {
 				log.Infof("Resetting the UserID of sideCar proxy as it matches with the app container for Pod %q", params.pod.Name)
 				sideCarProxy.SecurityContext.RunAsUser = nil
 				sideCarProxy.SecurityContext.RunAsGroup = nil
 			}
 		}
 	}
+
+	var nodes kclient.Client[*corev1.Node]
+	if wh.nodes != nil {
```

Comment:
Can we short-circuit these nil checks to clean the code up? If `wh.nodes == nil`, we should probably only enable native sidecars if it was explicitly set to true (and not defaulted). I think there's some code in the env package that enables checking this